source/geod24/LocalRest.d

```diff
@@ -2157,6 +2157,55 @@ unittest
     thread_joinAll();
 }
 
+/// Test restarting a node that has responses waiting for it
+unittest
+{
+    static interface API
+    {
+        @safe:
+        public void call0 ();
+        public void call1 ();
+    }
+
+    __gshared C.Tid node2Addr;
+
+    static class Node : API
+    {
+        @trusted:
+
+        public override void call0 ()
+        {
+            scope node2 = new RemoteAPI!API(node2Addr);
+            node2.call1();
+        }
+
+        public override void call1 ()
+        {
+            Thread.sleep(1.seconds);
```

Comment:
Use the `sleep` call, not `Thread.sleep`.
Also set a `__gshared bool hasRun` at the end so you can while / cas on it in the main thread, like I did with the network test yesterday (or just use a mutex with timeout...)

Reply:
Hmm, I think I will still need to add a little bit of thread.sleep here though. Because I don't know at which point the Response will be received by node 1. Consider for example:

```diff
diff --git a/source/geod24/LocalRest.d b/source/geod24/LocalRest.d
index a06bd63..8343c9e 100644
--- a/source/geod24/LocalRest.d
+++ b/source/geod24/LocalRest.d
@@ -2168,6 +2168,7 @@ unittest
     }

     __gshared C.Tid node2Addr;
+    __gshared bool hasWokenUp;

     static class Node : API
     {
@@ -2181,13 +2182,14 @@ unittest

         public override void call1 ()
         {
-            Thread.sleep(1.seconds);
+            hasWokenUp = true;
         }
     }

     auto node1 = RemoteAPI!API.spawn!Node(500.msecs);
     auto node2 = RemoteAPI!API.spawn!Node();
     node2Addr = node2.ctrl.tid();
+    node2.ctrl.sleep(2.seconds, false);

     try
     {
@@ -2198,8 +2200,7 @@ unittest

     node1.ctrl.restart();

-    // this will trigger the range error
-    Thread.sleep(2.seconds);
+    while (!hasWokenUp) {}

     node1.ctrl.shutdown();
     node2.ctrl.shutdown();
```


After the `while` loop there, we still can't be sure the `Response` was sent back yet after `call1` returned.

Or did you have another idea?